### PR TITLE
12.2.1 features

### LINKED ISSUE #448: macOS Ventura 13.1 Not Supported
### Description

When attempting to open the Transaction Tool a message is displayed saying, "TransactionTools is damaged and can't be opened. You should move it to the Trash". This message appears to be a macOS message box and is displayed before the tool runs, as no Transaction Tool log files are created or updated. This message appears after re-installing the Transaction Tool as well.

### Steps to reproduce

Install the Transaction Tool and attempt to run the App

### Additional context

_No response_

### Hedera network

other

### Version

v12.2

### Operating system

macOS

### PR DESCRIPTION
**Pull request by jbair06**
**Description**:
Due to the rollback, and some necessary fixes while trying to structure the updates, a number issues were resolved simultaneously. In addition to the fixes below, this PR also contains a seemingly large speed increase to the signing process when a large amount of transactions are present. A cache system was also implemented, allowing for automated archiving of transactions that are expired.

The cache system used is [Caffeine](https://github.com/ben-manes/caffeine) ([google's CacheBuilder named Caffeine as the successor](https://guava.dev/releases/31.0-jre/api/docs/com/google/common/cache/CacheBuilder.html#:~:text=Prefer%20Caffeine%20over%20Guava%27s%20caching%20API&text=Caffeine%20offers%20better%20performance%2C%20more,to%20use%20any%20Guava%20types.)).

**Related issue(s)**:

Fixes #448, #467, #468 as well as speed improvements.

**Notes for reviewer**:
This version of the tool is currently being manually tested by Simi and Mark.

### CODE REVIEW

File: tools-cli/src/main/java/com/hedera/hashgraph/client/cli/options/CollateCommand.java [hunk 1]
```diff
@@ -226,47 +221,69 @@ private List<List<String>> verifyTransactions() {
...
+			} catch (HederaClientRuntimeException e) {
+				// If collating failed, add an item to the verification list
+				// Make sure this transaction isn't already in the map
+				// (Multi-node submission will have duplicate transactionId entries)
+				if (!verifyWithFiles.containsKey(transactionId)) {
+					var verificationItemList = new ArrayList<String>();
+					verificationItemList.add(fileName);
+					verificationItemList.add(transactionId);
+					verifyWithFiles.put(transactionId, verificationItemList);
+					logger.info("Collation and Verification of " + transactionId
```
Comment (Neeharika-Sompalli) [inline tools-cli/src/main/java/com/hedera/hashgraph/client/cli/options/CollateCommand.java:244]: Do you think its an info level message or a warn ?

File: tools-cli/src/main/java/com/hedera/hashgraph/client/cli/options/CollateCommand.java [hunk 2]
```diff
@@ -226,47 +221,69 @@ private List<List<String>> verifyTransactions() {
...
+			if (!verifyWithFiles.containsKey(transactionId)) {
+				var verificationItemList = new ArrayList<String>();
+				verificationItemList.add(fileName);
+				verificationItemList.add(transactionId);
+				verificationItemList.add("\"" + String.join(",", requiredIdsInUse) + "\"");
+				verificationItemList.add("\"" + String.join(",", publicKeyNames) + "\"");
+
+				// Put the list of strings into the map
+				verifyWithFiles.put(transactionId, verificationItemList);
+			}
```
Comment (Neeharika-Sompalli) [inline tools-cli/src/main/java/com/hedera/hashgraph/client/cli/options/CollateCommand.java:284]: Seems this logic can be in a extracted to a method ? Seems almost similar to logic on 239

File: tools-ui/src/test/java/com/hedera/hashgraph/client/ui/utilities/UpdateHelperTest.java [hunk 3]
```diff
@@ -52,64 +54,54 @@ class UpdateHelperTest {
...
 
 	@AfterEach
-	void tearDown() throws IOException {
-		FileUtils.deleteDirectory(new File(DOCUMENTS, TRANSACTION_TOOLS));
-		if (new File(DOCUMENTS, TRANSACTION_TOOLS + "_test").exists()) {
-			FileUtils.moveDirectory(new File(DOCUMENTS, TRANSACTION_TOOLS + "_test"),
-					new File(DOCUMENTS, TRANSACTION_TOOLS));
-		}
-	}
+	void tearDown() {}
```
Comment (Neeharika-Sompalli) [inline tools-ui/src/test/java/com/hedera/hashgraph/client/ui/utilities/UpdateHelperTest.java:66]: Do we need this if its empty ?

File: tools-cli/src/main/java/com/hedera/hashgraph/client/cli/options/CollateCommand.java (hunk 1 above)
Comment (jbair06) [reply tools-cli/src/main/java/com/hedera/hashgraph/client/cli/options/CollateCommand.java]: Yes, that is probably better. Updated.

File: tools-ui/src/test/java/com/hedera/hashgraph/client/ui/utilities/UpdateHelperTest.java (hunk 3 above)
Comment (jbair06) [reply tools-ui/src/test/java/com/hedera/hashgraph/client/ui/utilities/UpdateHelperTest.java]: Indeed we do not. Updated

File: tools-cli/src/main/java/com/hedera/hashgraph/client/cli/options/CollateCommand.java (hunk 2 above)
Comment (jbair06) [reply tools-cli/src/main/java/com/hedera/hashgraph/client/cli/options/CollateCommand.java]: True, and updated.

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM ! Thanks @jbair06

**Pull request merged**